### ISSUE TITLE
TypeVariableSignature.resolve() throws "Class name is not set" for generic superclass signature arguments

### ISSUE THREAD

**Issue by platosha** [classgraph/classgraph#706]
I have a generic class that bypasses a type parameter to its generic superclass:

```java
public class GenericBypass<T> extends GenericBase<T> {
    public T bypassMethod() {
        return null;
    }
}
```

I am trying to call `TypeVariableSignature.resolve()` for the T argument in the superclass signature with the following code:

```java
var bypassCls = scanResult.getClassInfo(GenericBypass.class.getName());
var superclassArg = bypassCls.getTypeSignature().getSuperclassSignature().getTypeArguments().get(0);
var superclassArgTVar = (TypeVariableSignature) superclassArg.getTypeSignature();
var bypassTParamFromSuperclassArg = superclassArgTVar.resolve();
```

### Expected outcome

`TypeParameter` instance for T parameter of GenericBypass class is returned, equal to the `TypeParameter` T instance from resolving the T variable from `bypassMethod` result type signature. 

### Actual outcome

```
Exception in thread "main" java.lang.IllegalArgumentException: Class name is not set
	at io.github.classgraph.ScanResultObject.getClassInfo(ScanResultObject.java:118)
	at io.github.classgraph.TypeVariableSignature.resolve(TypeVariableSignature.java:100)
	at com.example.classgraph.Main.main(Main.java:23)
```

[verbose_log.txt](https://github.com/classgraph/classgraph/files/9658038/verbose_log.txt)
[classgraph-type-arguments-bypass.zip](https://github.com/classgraph/classgraph/files/9658042/classgraph-type-arguments-bypass.zip)

**Comment by lukehutch** [comment]
Hi @platosha,

I'm sorry it took me so long to fix this.

I believe the issue is fixed. Please try the latest git version.

**Comment by lukehutch** [comment]
Released in 4.8.150. Please let me know if there are still issues, and I'll reopen.
